django/tasks/base.py

```diff
@@ -0,0 +1,310 @@
+from dataclasses import dataclass, field, replace
+from datetime import datetime
+from inspect import isclass, iscoroutinefunction
+from typing import Any, Callable, Dict, Optional
+
+from asgiref.sync import async_to_sync, sync_to_async
+
+from django.db.models.enums import TextChoices
+from django.utils.module_loading import import_string
+from django.utils.translation import pgettext_lazy
+
+from .exceptions import TaskIntegrityError
+
+DEFAULT_TASK_BACKEND_ALIAS = "default"
+DEFAULT_QUEUE_NAME = "default"
+MIN_PRIORITY = -100
+MAX_PRIORITY = 100
+DEFAULT_PRIORITY = 0
+
+TASK_REFRESH_ATTRS = {
+    "errors",
+    "_return_value",
+    "finished_at",
+    "started_at",
+    "last_attempted_at",
+    "status",
+    "enqueued_at",
+    "worker_ids",
+}
+
+
+class ResultStatus(TextChoices):
+    READY = ("READY", pgettext_lazy("Task", "Ready"))
+    """The task has just been enqueued, or is ready to be executed again."""
+
+    RUNNING = ("RUNNING", pgettext_lazy("Task", "Running"))
+    """The task is currently running."""
+
+    FAILED = ("FAILED", pgettext_lazy("Task", "Failed"))
+    """
+    The task raised an exception during execution, or was unable to start.
+    """
+
+    SUCCEEDED = ("SUCCEEDED", pgettext_lazy("Task", "Succeeded"))
+    """The task has finished running successfully."""
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class Task:
+    priority: int
+    """The priority of the task"""
+
+    func: Callable
+    """The task function"""
+
+    backend: str
+    """The name of the backend the task will run on"""
+
+    queue_name: str = DEFAULT_QUEUE_NAME
+    """The name of the queue the task will run on"""
```

Comment:
I feel like this is inconsistent and I would be tempted to have 
```suggestion
    backend: str
    """The name of the backend the task will run on"""

    queue_name: str
    """The name of the queue the task will run on"""
```

Given `@task` has the defaults here (same with priority having the default)